### PR TITLE
Relax poc receipt submit timeouts and track transactions via callbacks

### PR DESCRIPTION
**Pull request by Vagabond**
If the submitting node is not more than 5 blocks behind the chain it
will never see its poc request hit the chain and thus end up in a loop
of never submitting receipts. This lengthens the poc request submission
timeout to the same as the poc interval and uses the transaction
callback to log if the receipt has been submitted.

### CODE REVIEW

File: src/poc/miner_poc_statem.erl [hunk 1]
```diff
@@ -206,6 +209,9 @@ mining(enter, _State, Data)->
     {keep_state, Data};
 mining(info, Msg, #data{blockchain=undefined}=Data) ->
     handle_event(info, Msg, Data);
+mining(info, {TxnRef, Result}, #data{txn_ref=TxnRef}=Data) ->
+    lager:info("PoC request submission result ~p", [Result]),
+    {keep_state, Data};
```
Comment (vihu) [inline src/poc/miner_poc_statem.erl:214]: If you got the result for the `txn_ref` why keep the ref in `Data` at this point?

Comment (Vagabond) [reply src/poc/miner_poc_statem.erl:214]: Why not?

Comment (vihu) [reply src/poc/miner_poc_statem.erl:214]: Hmm, I guess its fine since it will get a new ref on new submission anyway

**Comment by Vagabond** [comment]
Yeah, we can certainly be more sophisticated here, this was just to improve behaviour in the simplest way possible

**Pull request merged**